src/poly/commitment.rs

```diff
@@ -21,6 +23,95 @@ pub struct OpeningProof<C: CurveAffine> {
     z2: C::Scalar,
 }
 
+/// A multiscalar multiplication in the polynomial commitment scheme
+#[derive(Debug)]
+pub struct MSM<C: CurveAffine> {
```

Comment:
Sorry that I forgot to put this field in there before, but we should put a pointer to Params in MSM.

```rust
pub struct MSM<'a, C: CurveAffine> {
    params: &'a Params<C>,

    /// TODO: documentation
    g_scalars: Option<Vec<C::Scalar>>,

    /// TODO: documentation
    h_scalar: Option<C::Scalar>,

    /// TODO: documentation
    other_scalars: Vec<C::Scalar>,

    /// TODO: documentation
    other_bases: Vec<C>,
}
```

That way, whenever you call methods of `MSM` you don't have to provide the `Params`.